src/design-patterns/MazeGame/maze_game.hpp

```diff
@@ -0,0 +1,23 @@
+#ifndef MAZE_GAME_H
+#define MAZE_GAME_H
+
+#include <memory>
+
+#include "maze.hpp"
+#include "maze_factory.hpp"
+
+class MazeGame {
+ public:
+  MazeGame() = default;
+  ~MazeGame() = default;
+
+  MazeGame(const MazeGame&) = delete;             // copy constructor
+  MazeGame(MazeGame&&) = delete;                  // move constructor
+  MazeGame& operator=(const MazeGame&) = delete;  // copy assignment
+  MazeGame& operator=(MazeGame&&) = delete;       // move assignment
+
+  std::unique_ptr<Maze> CreateMaze();
+  std::unique_ptr<Maze> CreateMaze(MazeFactory& factory);
```

Comment:
MazeGame 내부에 maze, room, door 이런 것들이 생성되는 것이라면, 
MazeGame 객체 내에서 factory 관리해도 될 듯하다. 

```suggestion
  void CreateMaze();
  void CreateMaze (int num_maze);
private:
   std::vector<std::unique_ptr<Maze>> _maze_vec; 
   // 또는 std::unordered_map<int, std::unique_ptr<Maze>> _maze_map;
   MazeFactory _mfac;
   
   
   // cpp
   void MazeGame::CreateMaze (int num_maze){
     //num_maze 만큼
     _maze_vec.emplace_back (_mfactory.makeMaze ());
```

Reply:
Application 객체를 관리하게 될 때 이 방법이 좋겠다